src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -111,6 +111,7 @@ private List<SourceRepository> GetEffectiveSourcesCore(ISettings settings)
         {
             // Take the passed in sources
             var packageSources = new HashSet<string>(Sources, StringComparer.Ordinal);
+            var sourceObjects = new Dictionary<string, PackageSource>(StringComparer.Ordinal);
```

Comment:
I think packageSources should stay, the HashSet<string> is used to filter out dupes that come from different places (like FallbackSources). I'll chat with you about this.

---
In reply to: [62936465](https://github.com/NuGet/NuGet.Client/pull/586#discussion_r62936465) [](ancestors = 62936465)